### PR TITLE
Trust registry service account in STS policy

### PR DESCRIPTION
**Pull request by csrwng**
Fixes issue in which the image registry itself cannot access the bucket created by the registry operator

### CODE REVIEW

File: cmd/infra/aws/iam.go [hunk 1]
```diff
@@ -682,3 +667,60 @@ func iamTags(infraID, name string) []*iam.Tag {
...
+	}
+	if len(serviceAccounts) == 1 {
+		params.ServiceAccounts = fmt.Sprintf("%q", serviceAccounts[0])
+	} else {
+		sas := &bytes.Buffer{}
+		fmt.Fprintf(sas, "[")
+		for i, sa := range serviceAccounts {
+			fmt.Fprintf(sas, "%q", sa)
+			if i < len(serviceAccounts)-1 {
+				fmt.Fprintf(sas, ", ")
```
Comment (sjenning) [inline cmd/infra/aws/iam.go:710]: @csrwng just making sure you tested this.  I've never tried to put two SAs as the `sub`.

**Comment by csrwng** [comment]
I did test it. You can quickly try it out with `hypershift create iam aws`

**Pull request merged**